Server/JokesOnYou.Web.Api/Services/JokesService.cs

```diff
@@ -12,12 +14,58 @@ namespace JokesOnYou.Web.Api.Services
     public class JokesService : IJokesService
     {
         private readonly IJokesRepository _jokesRepo;
+        private readonly IUserRepository _userRepository;
+        private readonly IUnitOfWork _unitOfWork;
         private readonly IMapper _mapper;
 
-        public JokesService(IJokesRepository jokesRepo, IMapper mapper)
+        public JokesService(IJokesRepository jokesRepo, IUserRepository userRepository, IUnitOfWork unitOfWork, IMapper mapper)
         {
             _jokesRepo = jokesRepo;
             _mapper = mapper;
+            _userRepository = userRepository;
+            _unitOfWork = unitOfWork;
+        }
+
+        public async Task<JokeReplyDto> CreateJokeAsync(JokeCreateDto jokeCreateDto)
+        {
+            TrimAndNormalizeJokeCreateDto(jokeCreateDto);
+
+            var isDuplicate = await _jokesRepo.DoesJokeExist(jokeCreateDto);
```

Comment:
you can insert the await.... inside the if statement since it returns a boolean

Reply:
Yes its possible, I thought this might be more readable but I guess it could be readable enough.